pulp_python/app/serializers.py

```diff
@@ -226,20 +226,21 @@ def deferred_validate(self, data):
                     "The uploaded artifact's sha256 checksum does not match the one provided"
                 )}
             )
-        sha256 = artifact.sha256
-        if sha256 and python_models.PythonPackageContent.objects.filter(sha256=sha256).exists():
-            raise serializers.ValidationError(detail={"sha256": _('This field must be unique')})
 
         _data = parse_project_metadata(vars(metadata))
         _data['packagetype'] = metadata.packagetype
         _data['version'] = metadata.version
         _data['filename'] = filename
-        _data['sha256'] = sha256
+        _data['sha256'] = artifact.sha256
 
         data.update(_data)
 
         return data
 
+    def retrieve(self, validated_data):
+        content = python_models.PythonPackageContent.objects.filter(sha256=validated_data["sha256"])
+        return content.first()
```

Comment:
Can there be more than one? Should it be
```suggestion
        return python_models.PythonPackageContent.objects.get(sha256=validated_data["sha256"])
```

Reply:
There could be 1 or 0. If there is zero then `get` will raise an error while `first` will return `None`.

Reply:
And `None` is a valid response. :heavy_check_mark: